Git.hub/Client.cs

```diff
@@ -148,6 +148,20 @@ public User getCurrentUser()
             return user.Data;
         }
 
+        public User getUser(string username)
+        {
+            if (username == null)
+                throw new ArgumentNullException(nameof(username));
+
+            if (username.Length == 0)
+                throw new ArgumentException("Empty username", nameof(username));
+
+            var request = new RestRequest($"/users/{username}");
+
+            var user = client.Get<User>(request);
+            return user.Data;
```

Comment:
Should the method handle exceptions (e.g. connection timeout, host not found etc, 404)?

Reply:
I tried to follow existing code patterns. The `getCurrentUser` method only checks for null and assumes it's because of bad credentials (although it could in fact be anything). But this API doesn't require authentication, so I removed this check. If you like, I could throw a "Not found" exception if the user doesn't exist, but just returning null makes sense too. As for connection timeout, etc, the other methods don't do it, so I didn't think it would make sense to do it here.

Reply:
It doesn't seem the existing code makes itself concerned with error handling. So I suppose you just follow the existing patterns.
That is something we'll have to tackle separately